### ISSUE TITLE
Add functionality to display x-axis and y-axis

### ISSUE THREAD

**Issue by Sudha247** [Sudha247/ocaml-joy#13]
For debugging purposes, it's useful too see the cartesian axes. We can have a boolean reference on whether to display the axes, or not. We have to expose a function `display_axis : bool -> unit` where, `dispay_axis true` enables displaying the axes and `display_axis false` disables it.

Later in the init method, if the reference is true, we should display the axes.

https://github.com/Sudha247/ocaml-joy/blob/3b8dc79d4246f52a1905f8770d41d32f06eb9bb8/lib/shape.ml#L37-L39

The axes need to duller in colour, perhaps some shade of grey, to distinguish them from other shapes. 

Useful reading material: https://cs3110.github.io/textbook/chapters/mut/refs.html

**Comment by nikochiko** [comment]
@FayCarsons please go ahead!

**Comment by FayCarsons** [comment]
So, I'm very new to OCaml. I have experience with Rust and Clojure but this syntax is a bit foreign in places. 

Here's what I have so far (in shape.ml): 
`let axes_flag = ref false;
let draw_axes flag = 
  axes_flag := flag;

let render_axes = 
  moveto 0 0;
  lineto 0 500;
  moveto 0 0;
  lineto 500 0;`

and then in init:
`let init () =
  open_graph " 500x500";
  if !axes_flag then
    set_color (rgb 127 127 127);
    render_axes;
    
  set_color black`

and when I try to run dune build I get a syntax error at line 30 of shape.ml, after another function(render_shape) that was already there? Really confused as to what's going on. Examples have been working and I've been able to play around a bit. I added the signature of draw_line to shape.mli as well. 

Can anyone else see an issue here?

**Comment by Sudha247** [comment]
Hi @FayCarsons, I think what you're trying to do is correct. Note that `;` in OCaml is merely a separator between two statements. So, when you want to create a reference variable `axes_flag` and a function to set it, it needs to be

```ocaml
let axes_flag = ref false in
let draw_axes flag = 
  axes_flag := flag
in
...
``` 

In your code this is separated by `;`, so the compiler treats it as a single block of code. I presume hence the syntax error. Could you please try this?

**Comment by TitilayoAdeyemi** [comment]
Hello! Since this issue has not been assigned yet, can I be assigned this issue please?

**Comment by Sudha247** [comment]
Hi @TitilayoAdeyemi, thanks for your interest. @FayCarsons is working on this and has an open PR. I encourage you to take a look at other good-first-issues. :)

**Comment by TitilayoAdeyemi** [comment]
Hi Sudha. Oh I'm sorry, I didn't realize it was assigned already. I have checked all the issues and they've all been assigned as well.

**Comment by Sudha247** [comment]
Implemented in #17. I'm closing this. Thanks for the work @FayCarsons!

**Comment by Sudha247** [comment]
@TitilayoAdeyemi I'll try to create more issues today.
